src/snowflake/cli/plugins/nativeapp/run_processor.py

```diff
@@ -109,7 +117,11 @@ def _create_dev_app(self, diff: DiffResult) -> None:
                     return
 
                 except ProgrammingError as err:
-                    generic_sql_error_handler(err)
+                    if err.errno not in UPGRADE_RESTRICTION_CODES:
+                        generic_sql_error_handler(err)
+                    else:
+                        cc.warning(err.msg)
```

Comment:
I understand here we're going to print whatever gets sent back from us from the backend -- what is the user going to see if the app has been orphaned this way? Is it enough to understand what's happening, or is it just going to be something like `Application is not available` which is potentially confusing @sfc-gh-fcampbell?

Reply:
The message could be one of a few depending on the reason for the upgrade being denied. In the case that this PR is addressing, it's "Application is no longer available for use. Please contact the application provider for more details.".

To keep things consistent, I lifted this code from https://github.com/snowflakedb/snowflake-cli/blob/38a4d680fe62913940cb85fd4e8e1735e0eb3d70/src/snowflake/cli/plugins/nativeapp/run_processor.py#L328-L333, which handles the same case when trying to upgrade an application using a release directive. As part of the error messaging improvements I'll do (https://github.com/snowflakedb/snowflake-cli/pull/1245#discussion_r1654931887), I'll make this a more friendly message (and in a separate improvement I'll reduce the code duplication going on here)

Reply:
yeah we could improve this message to explain what happened, but since the actual response from Snowflake is a bit vague on the details, I wasn't sure how much we want to reveal to the user. Is it OK for us to tell them that the package was dropped?